src/libsystemd/sd-journal/compress.c

```diff
@@ -681,6 +684,9 @@ int compress_stream_lz4(int fdf, int fdt, uint64_t max_bytes) {
         if (r < 0)
                 return r;
 
+        if (ret_uncompressed_size)
+                *ret_uncompressed_size = total_in;
```

Comment:
`total_in` uses the wrong type. please fix at the same time. (and `total_out` too)